examples/guard_riverpod/lib/main.dart

```diff
@@ -2,22 +2,42 @@
 import 'package:beamer/beamer.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
-import 'package:guard_riverpod/router.dart';
+import 'package:guard_riverpod/locations.dart';
 
-void main() => runApp(ProviderScope(child: MyApp()));
+void main() {
+  final container = ProviderContainer();
+  runApp(UncontrolledProviderScope(container: container, child: MyApp(container)));
+}
 
-class MyApp extends ConsumerWidget {
+class MyApp extends StatelessWidget {
+  MyApp(this._rootContainer);
+
+  final ProviderContainer _rootContainer;
```

Comment:
Nice improvement !

By storing the container in the app, some people may use this accidentally it in the build method, when they actually should use `WidgetRef`.

One way to  mitigate this issue, would be to call `createDelegate` in `main` and pass the delegate into `MyApp`

Reply:
Yeah, it's missing encapsulation, that's a bad approach.

Updated the root widget to pass beamer-agnostic properties, which are the `MaterialApp.router` required properties only.

Thanks for the suggestion